src/cpp/session/modules/SessionRSConnect.cpp

```diff
@@ -181,21 +181,21 @@ class RSConnectPublish : public async_r::AsyncRProcess
 
          if (!reticulatePython.empty())
          {
-            // we found a Python version; forward it
-            core::system::setenv(&env, "RETICULATE_PYTHON", reticulatePython);
+            // we found a Python version; forward it to RETICULATE_PYTHON_FALLBACK
+            core::system::setenv(&env, "RETICULATE_PYTHON_FALLBACK", reticulatePython);
```

Comment:
Double checked again:

Let's say these are set to 2 different Pythons
<img width="727" alt="image" src="https://user-images.githubusercontent.com/7817881/155162508-32f8b420-efa3-4ffe-b7df-c75b0e7288b7.png">

In the console, note that `.rs.inferReticulatePython()` correctly prefers the python from `RETICULATE_PYTHON`
<img width="583" alt="image" src="https://user-images.githubusercontent.com/7817881/155162623-cd4eea32-b9d2-4ae6-9fe1-359ff66c2651.png">

In the Terminal: 
<img width="654" alt="image" src="https://user-images.githubusercontent.com/7817881/155166213-9c02b165-1066-41db-ad5b-d1cbe4d629d8.png">
Note that, because my RETICULATE_PYTHON points to a virtual env python, in the terminal, that virtual env is activated (even though I did not call activate), which python points to the correct python, and both RETICULATE_PYTHON and RETICULATE_PYTHON_FALLBACK system environment variables are set to the same value in the terminal (even though RETICULATE_PYTHON_FALLBACK is set to a different value in the R environment).